concrete/blocks/desktop_app_status/view.php

```diff
@@ -1,19 +1,18 @@
-<?php defined('C5_EXECUTE') or die("Access Denied."); ?>
-
 <?php
-if (version_compare($latest_version, APP_VERSION, '>')) {
+defined('C5_EXECUTE') or die('Access Denied.');
+
+/** @var Concrete\Core\Block\View\BlockView $view */
+/** @var string $current_version the currently installed concrete5 version */
+/** @var string $latest_version the last available concrete5 version */
+/** @var int $updates number of packages with new versions available */
+
+if (version_compare($latest_version, $current_version, '>')) {
     ?>
     <div class="alert alert-info">
-        <?php echo t('The latest version of concrete5 is <strong>%s</strong>. You are currently running concrete5 version <strong>%s</strong>.', $latest_version, APP_VERSION) ?>
+        <?php echo t('The latest version of concrete5 is <strong>%s</strong>. You are currently running concrete5 version <strong>%s</strong>.', $latest_version, $current_version) ?>
         <a class="pull-right btn btn-info btn-xs" href="<?php echo $view->url('/dashboard/system/update/update')?>"><?php echo t('Update')?></a>
     </div>
     <?php
-} elseif (version_compare(APP_VERSION, Config::get('concrete.version'), '>')) {
```

Comment:
Why is this part removed and shouldn't the "skip_core" config setting be used somewhere here?

Reply:
> Why is this part removed

@ramonleenders this part has been removed in commit 378137789d7789a439e37692f81fbcd130495ba6 (included in this PR) because `APP_VERSION` is defined as the value of `Config::get('concrete.version')` (see [here](https://github.com/concrete5/concrete5/blob/8.5.1/concrete/src/Foundation/Runtime/Boot/DefaultBooter.php#L329)), so it's useless to compare these two values.

> shouldn't the "skip_core" config setting be used somewhere here

Nope, the above check (`version_compare($latest_version, $current_version, '>')`) already uses `skip_core`, since:

- `$latest_version` is the result of `Update::getLatestAvailableVersionNumber()`
- `Update::getLatestAvailableVersionNumber()` uses `Update::getLatestAvailableUpdate()` to determine the last version
- `Update::getLatestAvailableUpdate()` uses the `skip_core` setting